src/ui_basic/dropdown.h

```diff
@@ -275,9 +287,78 @@ template <typename Entry> class Dropdown : public BaseDropdown {
 	                  type,
 	                  style,
 	                  button_style) {
+		set_handle_textinput();
 	}
 	~Dropdown() override {
-		entry_cache_.clear();
+		filtered_entries.clear();
+		unfiltered_entries.clear();
+	}
+
+	void clear_filter() override {
+		if (current_filter_.empty()) {
+			return;
+		}
+		current_filter_.clear();
+		if (has_selection()) {
+			Entry selected_entry = get_selected();
+			restore_filtered_list();
+			select(selected_entry);
+		} else {
+			restore_filtered_list();
+		}
+	}
+	void restore_filtered_list() {
+		clear_filtered_list();
+		for (auto& ee : unfiltered_entries) {
+			add_to_filtered_list(ee.name, ee.value, ee.img, false, ee.tooltip, ee.hotkey);
+		}
+	}
+
+	void save_selected_entry(uint32_t index) override {
+		selected_entry_ = index < filtered_entries.size() ? *filtered_entries[index] : Entry{};
+	}
+
+	bool handle_textinput(const std::string& input_text) override {
+		std::string lower = std::string(input_text);
+		lower[0] = std::tolower(lower[0]);
+		current_filter_ = current_filter_.append(lower);
+
+		apply_filter();
+
+		return true;
+	}
+
+	void apply_filter() override {
+		clear_filtered_list();
+		add_matching_entries();
+
+		if (filtered_entries.empty()) {
+			add_no_match_entry();
+		}
+		// force list to stay open even if mouse is now away due to smaller
+		// dropdown list because of applied filter
+		set_list_visibility(true);
+	}
+
+	void add_no_match_entry() {
+		// re-add initially selected entry with adapted texts to inform user
+		for (auto& x : unfiltered_entries) {
+			if (x.value == selected_entry_) {
+				add_to_filtered_list(_("No matches"), x.value, x.img, false,
```

Comment:
How about changing the text to `"No matches for XYZ"`? boost::format can do this. For coolness try to see if the Listselect supports styling this with an Italic font tag

Reply:
no, the message must be as short as possible, because it may cause the display-button to adapt its width if needed be. This may break layouts, especially because we have no control of the filter's length